### ISSUE TITLE
Add `no-coerce-boolean` rule

### ISSUE THREAD

**Issue by TrevorBurnham** [marcalexiei/eslint-plugin-zod#248]
### 🧩 Description

**Problem**

`z.coerce.boolean()` uses JavaScript's `Boolean()` constructor, which treats any non-empty string — including `"false"` — as `true`. This is almost always a bug when parsing URL query parameters or form data, which is the most common use case for `z.coerce.*` schemas.

```ts
// ❌ Bug: Boolean("false") === true
const schema = z.object({
  isUrgent: z.coerce.boolean().optional(),
});

schema.parse({ isUrgent: "false" });
// → { isUrgent: true }
```

**Proposed rule: `zod/no-coerce-boolean`**

Report an error whenever `z.coerce.boolean()` is used, with a suggestion to use a string transform instead:

```ts
// ✅ Correct
z.string().transform((v) => v === "true")

// or, to also accept actual booleans:
z.union([z.boolean(), z.string().transform((v) => v === "true")])
```

### ⚙️ Options

N/A

### ✅ Valid Code Examples

```typescript
const schema = z.object({
  isUrgent: z.string().transform((v) => v === "true"),
});
```

### ❌ Invalid Code Examples

```typescript
const schema = z.object({
  isUrgent: z.coerce.boolean().optional(),
});
```

**Comment by marcalexiei** [comment]
[`stringbool`](https://zod.dev/api?id=stringbool) is better suited for handling these scenarios. 

However, given the variety of possible use cases, it may be more appropriate to simply flag the use of `z.coerce.boolean` without providing an automatic fix or recommendation.

FYI I've seen some APIs dealing working with `true` and `false` as string so this rule will not be included in the recommended configs.
